lib/idempotency/rails.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'base64'
+require_relative 'constants'
+
+module Idempotency
+  module Rails
+    def use_cache(request_identifiers = [], lock_duration: nil) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity
+      return unless cache_request?(request)
+
+      cache = Idempotency.cache
+
+      idempotency_key = request.headers[Constants::HEADER_KEY] || SecureRandom.hex
```

Comment:
Based on the RFC, the format is `X-Idempotency: "<key>"`, so we may need to remove the `"` part.